sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubProducerClient.cs

```diff
@@ -1111,6 +1134,27 @@ public virtual async ValueTask DisposeAsync()
                         eventData.ClearPublishingState();
                     }
 
+                    // Reset the partition state and options to ensure that future attempts
+                    // are safe and do not risk data loss by reusing the same producer group identifier.
+
+                    if (!Options.PartitionOptions.TryGetValue(options.PartitionId, out var partitionOptions))
+                    {
+                        partitionOptions = new PartitionPublishingOptionsInternal();
+                        Options.PartitionOptions[options.PartitionId] = partitionOptions;
+                    }
+
+                    partitionOptions.ProducerGroupId = null;
+                    partitionOptions.OwnerLevel = null;
+                    partitionOptions.StartingSequenceNumber = null;
+
+                    partitionState.ProducerGroupId = null;
```

Comment:
Would it be worth extracting into a method?

Reply:
maybe?   It's a pretty straightforward block used in two spots.   I'll give it a think.